patches/third_party-blink-renderer-platform-plugins-plugin_data.cc.patch

```diff
@@ -0,0 +1,14 @@
+diff --git a/third_party/blink/renderer/platform/plugins/plugin_data.cc b/third_party/blink/renderer/platform/plugins/plugin_data.cc
+index 8e1cba5dcfe9b753ca4c872b551e13b12b7d5f2e..0f9d0db71a4b759f19cb26848cf6a937474021cb 100644
+--- a/third_party/blink/renderer/platform/plugins/plugin_data.cc
++++ b/third_party/blink/renderer/platform/plugins/plugin_data.cc
+@@ -101,6 +101,9 @@ void PluginData::UpdatePluginList(const SecurityOrigin* main_frame_origin) {
+   Vector<mojom::blink::PluginInfoPtr> plugins;
+   registry->GetPlugins(false, main_frame_origin_, &plugins);
+   for (const auto& plugin : plugins) {
++    if (plugin->name != "Shockwave Flash") {
```

Comment:
if i understand correctly, this will never add any plugins to navigator.plugins/mimeTypes unless it is the Flash plugin. but the behavior we want is:

1. if flash is not enabled for the site via click-to-play, navigator.plugins/mimeTypes is empty
2. if flash is enabled, navigator.plugins/mimeTypes only shows Flash

i will update  https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode#privacy-protection-enabled-regardless-of-whether-fingerprinting-protection-mode-is-on to be more clear about this

Reply:
@diracdeltas - `GetPlugins` does not return the flash plugin if its not enabled via click-to-play. If flash is not enabled via click-to-play, navigator.plugins/mimeTypes will return empty.

Reply:
Resolved after offline conversation.